frameworks/cra/src/preset.ts

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable no-param-reassign */
+
+import { logger } from '@storybook/node-logger';
+import { dirname, join, relative } from 'path';
+import type { PresetProperty, Options } from '@storybook/core-common';
+import type { Configuration, RuleSetRule } from 'webpack';
+import type { FrameworkOptions, StorybookConfig } from './types';
+import { getReactScriptsPath } from './utils/getReactScriptsPath';
+import { testMatch } from './utils/testMatch';
+import { checkForIncompatiblePresets } from './utils/compatibility';
+
+export const addons: PresetProperty<'addons', StorybookConfig> = [
+  dirname(require.resolve(join('@storybook/preset-react-webpack', 'package.json'))),
+  dirname(require.resolve(join('@storybook/react', 'package.json'))),
+];
+
+const defaultFrameworkOptions: FrameworkOptions = {
+  legacyRootApi: true,
+};
+
+export const frameworkOptions = async (
+  _: never,
+  options: Options
+): Promise<StorybookConfig['framework']> => {
+  const config = await options.presets.apply<StorybookConfig['framework']>('framework');
+
+  if (typeof config === 'string') {
+    return {
+      name: config,
+      options: defaultFrameworkOptions,
+    };
+  }
+  if (typeof config === 'undefined') {
+    return {
+      name: require.resolve('@storybook/cra') as '@storybook/cra',
+      options: defaultFrameworkOptions,
+    };
+  }
+
+  return {
+    name: config.name,
+    options: {
+      ...defaultFrameworkOptions,
+      ...config.options,
+    },
+  };
+};
+
+export const core: PresetProperty<'core', StorybookConfig> = async (config, options) => {
+  const framework = await options.presets.apply<StorybookConfig['framework']>('framework');
+
+  return {
+    ...config,
+    builder: {
+      name: dirname(
+        require.resolve(join('@storybook/builder-webpack5', 'package.json'))
+      ) as '@storybook/builder-webpack5',
+      options: typeof framework === 'string' ? {} : framework.options.builder || {},
+    },
+  };
+};
+
+export const webpackFinal: StorybookConfig['webpack'] = async (config, options) => {
+  config.resolve = config.resolve || {};
+  config.module = config.module || {};
+  config.module.rules = config.module.rules || [];
+
+  checkForIncompatiblePresets(options);
+
+  const { presets, configDir } = options;
+  const frameworkOptions = await presets.apply<FrameworkOptions>('frameworkOptions');
+
+  const scriptsPath = await getReactScriptsPath(frameworkOptions.scriptsPackageName, options);
+  const CWD = process.cwd();
+
+  const craWebpackConfigPath = join(scriptsPath, 'config', 'webpack.config');
+  logger.info(`=> Loading Webpack configuration from '${relative(CWD, craWebpackConfigPath)}'`);
+
+  // eslint-disable-next-line import/no-dynamic-require, global-require
+  const craWebpackConfig: Configuration = require(craWebpackConfigPath)(config.mode);
+
+  config.module.rules = [
+    ...((config.module?.rules ?? []) as unknown as RuleSetRule[]).filter(
```

Comment:
Not according to TypeScript, technically it could be undefined at runtime.

Reminder that users/presets are not required to pass the mutated config down, they can return something new that has these properties missing!